File: nordeus_challenge_backend_spring_boot/src/main/java/com/mmileticc/nordeuschallenge/model/dto/RunEncounterDTO.java
package com.mmileticc.nordeuschallenge.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RunEncounterDTO {

    private int encounterIndex;
    private Long monsterId;
    private String monsterName;
    private int difficultyRank;
    private int maxHealth;
    private int attack;
    private int defense;
    private int magic;
    private List<MoveDTO> moves;
    private boolean defeated;
    private boolean unlocked;
}

File: Gauntlet of the Hero - GOTH/Assets/_Scripts/Models/Backend/RunEncounterDto.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace GauntletOfTheHero.Backend.Models
{
    [System.Serializable]
    public class RunEncounterDto
    {
        [JsonProperty("encounterIndex")]
        public int encounterIndex;

        [JsonProperty("monsterId")]
        public long monsterId;

        [JsonProperty("monsterName")]
        public string monsterName;

        [JsonProperty("difficultyRank")]
        public int difficultyRank;

        [JsonProperty("maxHealth")]
        public int maxHealth;

        [JsonProperty("attack")]
        public int attack;

        [JsonProperty("defense")]
        public int defense;

        [JsonProperty("magic")]
        public int magic;

        [JsonProperty("moves")]
        public List<MoveDto> moves = new List<MoveDto>();

        [JsonProperty("defeated")]
        public bool defeated;

        [JsonProperty("unlocked")]
        public bool unlocked;
    }
}